go/libraries/doltcore/migrate/transform.go

```diff
@@ -228,6 +232,12 @@ func migrateTable(ctx context.Context, table *doltdb.Table, new *doltdb.DoltDB)
 	if err != nil {
 		return nil, err
 	}
+	if sch, err = patchMigrateSchema(ctx, sch); err != nil {
+		return nil, err
+	}
+	if err = validateSchema(sch); err != nil {
```

Comment:
can `validateSchema` ever fail? the patch code looks like it plugs the gaps with the same comparison condition